src/shared_utilities/general_utils.jl

```diff
@@ -0,0 +1,18 @@
+export searchsortednearest
```

Comment:
not sure, should this just go into src/shared_utilities/utils.jl instead of making a new file? Perhaps a future PR can be to split utils.jl into more descriptive pieces. I can open an issue if we want to do this.

Reply:
The `utils.jl` file has of ClimaLSM-specific functions, where this one is very general (hence `general_utils`). I don't want to import `utils.jl` in the `TimeVaryingInput` module  because it brings a lot of stuff I don't need.

Reply:
this is fine! we can clean up our utils (callback_utils.jl might be its own thing) later.

Reply:
or we can move the function into TimeVaryingInput if that makes sense (if other parts of ClimaLSM do not need it). not a big deal either way I think

Reply:
I will keep it there for the time being. I think it is a useful function to have around.